### PR TITLE
NoSuchElementException with probe supervisor in TestActorRef

### LINKED ISSUE #21770: NoSuchElementException when using TestProbe as supervisor in TestActorRef
I've encountered an issue with the `TestActorRef` in a test where I use a `TestProbe` as the parent/supervisor. The Actor wrapped in the `TestActorRef` throws an Exception in a certain case and this causes the supervisor strategy 
This used the work in 2.4.8 and now that I have updated to release 2.4.11 i get an `NoSuchElementException`.

To me it seems that this is introduced by the addition of the `DelegatingSupervisorStrategy` in `TestKit.scala` as part of PR #21201 where the overriden methods `processFailure` and `handleFailure` directly access the `delegates` Map. It seems that the method `delegate` should be used, as it is not used anywhere else.

To reproduce:

```
import akka.actor.{Actor, ActorSystem, Props}
import akka.testkit.{ImplicitSender, TestActorRef, TestKit, TestProbe}
import org.scalatest.WordSpecLike

class MinimalTestCase extends TestKit(ActorSystem("Testing")) with ImplicitSender with WordSpecLike {

  "A TestActorRef" should {
    "not throw an exception when parent is passed in the apply" in {
      val parent = TestProbe()
      val child = TestActorRef(Props(new Actor {
        def receive: Receive = {
          case x => throw new IllegalArgumentException("Throw me")
        }
      }), parent.ref, "Child")

      child ! 1
    }
    "not throw an exception when child is created through childActorOf" in {
      val parent = TestProbe()
      val child = parent.childActorOf(Props(new Actor {
        def receive: Receive = {
          case x => throw new IllegalArgumentException("Throw me")
        }
      }), "Child")

      child ! 1
    }
  }
}
```

Truncated stacktrace:
```
[ERROR] [10/31/2016 07:41:58.404] [Testing-akka.actor.default-dispatcher-5] [akka://Testing/system/testProbe-2] key not found: TestActor[akka://Testing/system/testProbe-2/Child]
java.util.NoSuchElementException: key not found: TestActor[akka://Testing/system/testProbe-2/Child]
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at akka.testkit.TestActor$DelegatingSupervisorStrategy.handleFailure(TestKit.scala:84)
	at akka.actor.dungeon.FaultHandling$class.handleFailure(FaultHandling.scala:263)
	at akka.actor.ActorCell.handleFailure(ActorCell.scala:374)
...
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #21770

### CODE REVIEW

File: akka-testkit/src/main/scala/akka/testkit/TestKit.scala [hunk 1]
```diff
@@ -77,11 +77,11 @@ object TestActor {
     }
 
     override def processFailure(context: ActorContext, restart: Boolean, child: ActorRef, cause: Throwable, stats: ChildRestartStats, children: Iterable[ChildRestartStats]): Unit = {
-      delegates(child).processFailure(context, restart, child, cause, stats, children)
+      delegate(child).processFailure(context, restart, child, cause, stats, children)
```
Comment (drewhk) [inline akka-testkit/src/main/scala/akka/testkit/TestKit.scala]: er, wat? Why does this fix the issue, lol :D

Comment (johanandren) [inline akka-testkit/src/main/scala/akka/testkit/TestKit.scala]: the delegate method handles missing entry, apply on map does not and gives NPE

Comment (johanandren) [inline akka-testkit/src/main/scala/akka/testkit/TestKit.scala]: oh, not NPE, NoSuchElementException i meant

Comment (drewhk) [inline akka-testkit/src/main/scala/akka/testkit/TestKit.scala]: Thanks

**Review by drewhk — APPROVED** [review]

**Pull request merged**